Content.Server/Abilities/Mime/MimePowersSystem.cs

```diff
@@ -146,6 +156,8 @@ public void RetakeVow(EntityUid uid, MimePowersComponent? mimePowers = null)
             mimePowers.ReadyToRepent = false;
             mimePowers.VowBroken = false;
             AddComp<MutedComponent>(uid);
+            if (mimePowers.PreventWriting)
+                AddComp(uid, new IlliterateComponent { FailWriteMessage = mimePowers.FailWriteMessage });
```

Comment:
This should be EnsureComp to make sure it doesn't try to add a new comp to a mob that somehow alredy has it, through dev shenanigans or something